chemprop/train/make_predictions.py

```diff
@@ -129,15 +129,15 @@ def make_predictions(args: PredictArgs, smiles: List[List[str]] = None) -> List[
     # Get prediction column names
     if args.dataset_type == 'multiclass':
         task_names = [f'{name}_class_{i}' for name in task_names for i in range(args.multiclass_num_classes)]
-    else:
-        task_names = task_names
 
     # Copy predictions over to full_data
     for full_index, datapoint in enumerate(full_data):
         valid_index = full_to_valid_indices.get(full_index, None)
-        preds = avg_preds[valid_index] if valid_index is not None else ['Invalid SMILES'] * len(task_names)
+        preds = avg_preds[valid_index] if valid_index is not None else ['Invalid SMILES'] * num_tasks
```

Comment:
Is num_tasks and len(task_names) equivalent? I am especially thinking about edge_cases, where e.g. one column was excluded by chemprop because it didn't hold a single label.

Reply:
I don't think I see the place where an empty column would lead to that changing. We skip empty rows but do we skip empty columns?

This is an excellent note. I thought that these were equivalent, and they are at the start. `args.num_tasks` is not actually a normal attribute, it's a property that is just `len(args.task_names)`. However, it looks like the num_targets variable is updated when it's a multiclass job (my other problem!) so I need to take a closer look at this.